src/libPMacc/include/random/distributions/uniform/Uniform_float.hpp

```diff
@@ -48,10 +54,125 @@ namespace detail
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne<float>::SwapOneToZero,
+        T_RNGMethod,
+        void
+    >
+    {
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+    public:
+        typedef float result_type;
+
+        DINLINE result_type
+        operator()(StateType& state)
+        {
+            const result_type randomValue =
+                PMacc::random::distributions::Uniform<
+                    uniform::ExcludeZero<result_type>
+            >()(state);
+            return randomValue == 1.0f ? 0.0f : randomValue;
+        }
+    };
+
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * Reduce the random range to 2^24.
+     * Creates intervals with the width of epsilon/2.
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne<float>::Use24Bit,
+        T_RNGMethod,
+        void
+    >
+    {
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+    public:
+        typedef float result_type;
+
+        DINLINE result_type
+        operator()(StateType& state)
+        {
+            const result_type 2pow24Inv = 5.9604645e-08f;
+            const randomValue24Bit = RNGMethod().get32Bits(state) >> 8;
+            return static_cast<result_type>( randomValue24Bit ) * PMACC_RNG_2POW24_INV;
+        }
+    };
+
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * Loops until a random value inside the defined range is created.
+     * The runtime of this method is not deterministic.
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne<float>::Repeat,
+        T_RNGMethod,
+        void
+    >;
+    {
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+    public:
+        typedef float result_type;
+
+        DINLINE result_type
+        operator()(StateType& state)
+        {
+            do
+            {
+                const result_type randomValue =
+                    PMacc::random::distributions::Uniform<
+                        uniform::ExcludeZero<result_type>
+                    >()(state);
+                if( randomValue != 1.0f )
+                    return randomValue;
+            }
+            while(true);
+        }
+    };
+
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * The default case use the method
+     * `uniform::::ExcludeOne<float,uniform::Use24Bit>`
+     */
+    template<class T_RNGMethod>
+    class Uniform<float, T_RNGMethod, void> :
+        public PMacc::random::distributions::Uniform<
+            uniform::ExcludeOne<float>::Use24Bit
+        >
+    {
+    };
+
+    /** Returns a random float value uniformly distributed in [0,1)
+     *
+     * The default case  for `ExcludeOne` use the method
```

Comment:
Here should be only `Equivalent to uniform::ExcludeOne<float>::Use24Bit` or similar to above (freedom to choose implementation).`Default case` is meaningless here. This should go to the `Uniform` template